AWS IAM Profile credentials (#210)
exasol/spark-connector: doc/user_guide/user_guide.md

spark-shell --jars spark-connector-s3_2.13-2.0.0-spark-3.4.1-assembly.jar
```

## AWS Authentication

If S3 intermediate storage is used, proper AWS Authentication parameters has to be provided:

* Spark has to be able to read and write into S3 (to export and import dataframe's data);
* Database has to be able to read and write into S3 (to perform `IMPORT` and `EXPORT` statements).

There are several ways to provide AWS credentials and concrete method depends on configuration of your cloud infrastructure. Here we cover main scenarios and configuration options you can tweak.

### Credential Providers

> The first option is `awsCredentialsProvider` with which you can specify list of ways credentials are retrieved from your spark environment. This parameter is not required and if not specified, default list of credentials providers are being used. At the moment of writing, this list includes the following credentials providers:

Reviewer: ```suggestion
The first option is `awsCredentialsProvider` with which you can specify list of ways credentials are retrieved from your spark environment. This parameter is not required and if not specified, the default list of credentials providers is being used. At the moment of writing, this list includes the following credentials providers:
```

* `org.apache.hadoop.fs.s3a.SimpleAWSCredentialsProvider`: credentials are explicitly set with options `awsAccessKeyId` and `awsSecretAccessKey`.
* `com.amazonaws.auth.EnvironmentVariableCredentialsProvider`: credentials are retrieved from environment variables `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` (of Spark process).
* `com.amazonaws.auth.InstanceProfileCredentialsProvider`: credentials are retrieved from EC2 instance IAM role.

> There are many more other credential providers in Amazon Hadoop library and 3rd party libraries. If you need to change default behaviour, you can set `awsCredentialsProvider` option to list of comma-separated classes.

Reviewer: ```suggestion
There are man other credential providers in Amazon Hadoop library and 3rd party libraries. If you need to change default behaviour, you can set `awsCredentialsProvider` option to a list of comma-separated class names.
```
